lib/pavilion/series.py

```diff
@@ -36,12 +43,173 @@ def test_obj_from_id(pav_cfg, test_ids):
 
 
+class SeriesManager:
+    """Series Manger"""
+
+    def __init__(self, _pav_cfg, _series_obj, _series_cfg):
+        self.pav_cfg = _pav_cfg
+        self.series_obj = _series_obj
+        self.series_cfg = _series_cfg
+
+        self.sets = self.series_cfg['series']
+
+        self.dep_graph = {}  # { set_name: [set_names it depends on] }
+        self.make_dep_graph()
+
+        sets_args = {}
+        universal_modes = self.series_cfg['modes']
+        # set up sets_args dict
+        for set_name, set_info in self.sets.items():
+
+            set_modes = set_info['modes']
+            all_modes = universal_modes + set_modes
+
+            args_list = ['run', '--series-id={}'.format(self.series_obj.id)]
+            for mode in all_modes:
+                args_list.append('-m{}'.format(mode))
+            args_list.extend(set_info['test_names'])
+
+            sets_args[set_name] = args_list
+
+        # create TestSet obj for each set
+        self.test_sets = {}
+        for set_name in sets_args:
+            self.test_sets[set_name] = TestSet(
+                self.pav_cfg, set_name, sets_args[set_name],
+                [], []
+            )
+
+        # create doubly linked graph
+        for set_name in self.dep_graph:
+            prev_str_list = self.dep_graph[set_name]
+            for prev in prev_str_list:
+                self.test_sets[set_name].add_prev(self.test_sets[prev])
+
+            next_str_list = []
+            for s_n in self.dep_graph:
+                if set_name in self.dep_graph[s_n]:
+                    next_str_list.append(s_n)
+
+            for next in next_str_list:
+                self.test_sets[set_name].add_next(self.test_sets[next])
+
+        # CLEAN UP ~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~
+
+        self.print_stats()
+
+        # kick off tests that aren't waiting on anyone
+        self.currently_running = []
+        for set_name in self.test_sets:
+            if not self.test_sets[set_name].get_prev():
+                self.test_sets[set_name].run_set()
+                self.currently_running.append(self.test_sets[set_name])
+
+        self.print_stats()
+
+        # keep checking on statuses of currently running sets
+        for set_obj in self.currently_running:
+            time.sleep(5)
+            if self.is_set_done(set_obj):
+                set_obj.change_stat('DONE')
+                for waiting in set_obj.get_next():
+                    run_this = True
+                    for prev in waiting.get_prev():
+                        if prev.get_stat() != 'DONE':
+                            run_this = False
+                    if run_this:
+                        waiting.run_set()
+                        self.currently_running.append(waiting)
+
+            self.print_stats()
+
+
+        # END CLEAN UP
+
+
+
+    def is_set_done(self, set_obj):
+        for test in set_obj.test_runs:
+            if not pathlib.Path(test.path/'RUN_COMPLETE').exists():
```

Comment:
test.path will always be a Path object, so you can just: `(test.path/RUN_COMPLETE).exists()`